### PR TITLE
R4R: update pricefeed module rest endpoints

### PR DESCRIPTION
**Pull request by denalimarsh**
- Get price: `curl http://127.0.0.1:1317/pricefeed/price/xrp:usd | jq`
- Get raw prices: `curl http://127.0.0.1:1317/pricefeed/rawprices/xrp:usd | jq`
- Get markets: `curl http://127.0.0.1:1317/pricefeed/markets | jq`
- Get oracles: `curl http://127.0.0.1:1317/pricefeed/oracles/xrp:usd | jq`
- Get params: `curl http://127.0.0.1:1317/pricefeed/parameters | jq`
- Post price: see `contrib/README.md`

### CODE REVIEW

File: x/pricefeed/client/rest/query.go [hunk 1]
```diff
@@ -14,17 +14,26 @@ import (
...
+		paramMarketID := vars[RestMarketID]
+		queryRawPricesParams := types.NewQueryWithMarketIDParams(paramMarketID)
+
+		bz, err := cliCtx.Codec.MarshalJSON(queryRawPricesParams)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryRawPrices), bz)
```
Comment (karzak) [inline x/pricefeed/client/rest/query.go]:
```suggestion
		res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryRawPrices), bz)
```

And then `cliCtx = cliCtx.WithHeight(height)` before post processing the response.

File: x/pricefeed/client/rest/query.go [hunk 2]
```diff
@@ -33,11 +42,30 @@ func queryRawPricesHandler(cliCtx context.CLIContext) http.HandlerFunc {
...
+		paramMarketID := vars[RestMarketID]
+		queryPriceParams := types.NewQueryWithMarketIDParams(paramMarketID)
+
+		bz, err := cliCtx.Codec.MarshalJSON(queryPriceParams)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryPrice), bz)
```
Comment (karzak) [inline x/pricefeed/client/rest/query.go]: see above

File: x/pricefeed/client/rest/query.go [hunk 3]
```diff
@@ -33,11 +42,30 @@ func queryRawPricesHandler(cliCtx context.CLIContext) http.HandlerFunc {
...
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		rest.PostProcessResponse(w, cliCtx, res)
+	}
+}
+
+func queryMarketsHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryMarkets), nil)
```
Comment (karzak) [inline x/pricefeed/client/rest/query.go]: same

File: x/pricefeed/client/rest/query.go [hunk 4]
```diff
@@ -46,9 +74,19 @@ func queryCurrentPriceHandler(cliCtx context.CLIContext) http.HandlerFunc {
...
+		paramMarketID := vars[RestMarketID]
+		queryOraclesParams := types.NewQueryWithMarketIDParams(paramMarketID)
+
+		bz, err := cliCtx.Codec.MarshalJSON(queryOraclesParams)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryOracles), bz)
```
Comment (karzak) [inline x/pricefeed/client/rest/query.go]: same

File: x/pricefeed/client/rest/tx.go [hunk 5]
```diff
@@ -5,23 +5,26 @@ import (
...
-	"github.com/gorilla/mux"
-
 	"github.com/cosmos/cosmos-sdk/client/context"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/rest"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
-	"github.com/kava-labs/kava/x/pricefeed/types"
+	"github.com/gorilla/mux"
+
+	// 	authclient "github.com/cosmos/cosmos-sdk/x/auth/client" TODO: use instead of utils?
```
Comment (karzak) [inline x/pricefeed/client/rest/tx.go]: remove TODO, `utils` is correct import

File: x/pricefeed/client/rest/query.go (hunk 1 above)
Comment (denalimarsh) [reply x/pricefeed/client/rest/query.go]: https://github.com/Kava-Labs/kava/pull/359/commits/d6df1fb7305aec68497e06464e15f6ae634eb619

File: x/pricefeed/client/rest/query.go (hunk 2 above)
Comment (denalimarsh) [reply x/pricefeed/client/rest/query.go]: https://github.com/Kava-Labs/kava/pull/359/commits/d6df1fb7305aec68497e06464e15f6ae634eb619

File: x/pricefeed/client/rest/query.go (hunk 3 above)
Comment (denalimarsh) [reply x/pricefeed/client/rest/query.go]: https://github.com/Kava-Labs/kava/pull/359/commits/d6df1fb7305aec68497e06464e15f6ae634eb619

File: x/pricefeed/client/rest/query.go (hunk 4 above)
Comment (denalimarsh) [reply x/pricefeed/client/rest/query.go]: https://github.com/Kava-Labs/kava/pull/359/commits/d6df1fb7305aec68497e06464e15f6ae634eb619

File: x/pricefeed/client/rest/tx.go (hunk 5 above)
Comment (denalimarsh) [reply x/pricefeed/client/rest/tx.go]: https://github.com/Kava-Labs/kava/pull/359/commits/d6df1fb7305aec68497e06464e15f6ae634eb619

**Pull request merged**